### ISSUE TITLE
App using Flutter 3.10 is more janky on iOS than previous versions.

### ISSUE THREAD

**Issue by Livinglist** [flutter/flutter#126575]
## Details

After upgrading to Flutter 3.10 for my app [Hacki](https://github.com/Livinglist/Hacki), there's obviously more janks than previous versions without impeller. (on iOS, I haven't tried for Android)

You can see for yourself by comparing the performance between [master branch](https://github.com/Livinglist/Hacki/tree/master) and [flutter-upgrade branch](https://github.com/Livinglist/Hacki/tree/flutter-3.10-upgrade)

The app itself doesn't need any extra configurations, you can just clone it and run it with `flutter run`.

| Flutter 3.7.11 | Flutter 3.10 |
| ------------- | ------------- |
| <video src="https://github.com/flutter/flutter/assets/7277662/47964931-0773-4bfe-b3eb-590575c25de5"> | <video src="https://github.com/flutter/flutter/assets/7277662/d9e950dc-3546-42f8-87c5-e7531fc443b9"> |

**Target Platform:** iOS
**Target OS version/browser:** iOS 16.4
**Devices:** iPhone 14

## Logs

<details>
<summary>Logs</summary>

```
```

```
[!] Flutter (Channel stable, 3.10.0, on macOS 13.2.1 22D68 darwin-arm64
    (Rosetta), locale en)
    ! Warning: `flutter` on your path resolves to
      /Users/jiaqifeng/Dev/flutter/bin/flutter, which is not inside your current
      Flutter SDK checkout at /Users/jiaqifeng/dev/flutter. Consider adding
      /Users/jiaqifeng/dev/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /opt/homebrew/Cellar/dart/2.17.0/libexec/bin/dart, which is not inside
      your current Flutter SDK checkout at /Users/jiaqifeng/dev/flutter.
      Consider adding /Users/jiaqifeng/dev/flutter/bin to the front of your
      path.
[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.78.1)
[✓] Connected device (3 available)
[✓] Network resources

```

</details>

**Comment by Livinglist** [comment]
Hi @jonahwilliams, I just attached two clips showcasing the difference. Hopefully it helps.

**Comment by jonahwilliams** [comment]
Ahh yeah, if I scroll it enough on 3.10 I can trigger some frame back ups. I belive this is fixed in the 3.11 beta, and at ToT I can no longer reproduce it. Please give it a shot if you have the time

**Comment by darshankawar** [comment]
@Livinglist 
Please check on latest beta as stated above and confirm.

**Comment by Livinglist** [comment]
looks like the issue is fixed in 3.11 beta, thanks!
